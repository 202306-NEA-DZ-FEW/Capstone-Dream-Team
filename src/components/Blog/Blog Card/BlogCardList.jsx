import React from "react";
import BlogCard from "./BlogCard";

export default function BlogCardList() {
    return (
        <div className='px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl md:px-4 lg:px-8 lg:py-20'>
            <div className='grid gap-5 lg:grid-cols-3 sm:max-w-sm sm:mx-auto lg:max-w-full'>
                <BlogCard />
            </div>
        </div>
    );
}
