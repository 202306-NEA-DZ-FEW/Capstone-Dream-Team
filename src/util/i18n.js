import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
