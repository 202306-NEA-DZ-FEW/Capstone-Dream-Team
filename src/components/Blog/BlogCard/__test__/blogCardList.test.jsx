import { I18nextProvider } from "react-i18next";
import renderer from "react-test-renderer";

import BlogCardList from "../blogCardList";
import i18n from "../../../../util/i18n";
const blogs = [
    {
        id: "1",
        data: {
            title: "Sample Title 1",
            content: "Sample Content 1",
            publish_date: "2023-10-30",
        },
    },
    {
        id: "2",
        data: {
            title: "Sample Title 2",
            content: "Sample Content 2",
            publish_date: "2023-10-31",
        },
    },
];
it("renders correctly", () => {
    const tree = renderer
        .create(
            <I18nextProvider i18n={i18n}>
                <BlogCardList blogs={blogs} numToShow={2} language='en' />
            </I18nextProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
