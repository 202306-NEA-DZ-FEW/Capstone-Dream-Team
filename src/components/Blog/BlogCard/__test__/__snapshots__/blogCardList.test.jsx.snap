// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="px-2 py-6 flex justify-center sm:max-w-xl md:max-w-full lg:max-w-full md:px-12 lg:px-2"
>
  <div
    className="grid gap-8 lg:grid-cols-3 md:grid-cols-2 sm:max-w-sm lg:max-w-full md:max-w-full mx-auto justify-center"
  >
    <div>
      <div
        className="dark:text-white flex flex-col justify-center lg:py-5 md:py-5 overflow-hidden w-80  transition-shadow duration-300 rounded group "
      >
        <div
          className="bg-white rounded-xl"
        >
          <a
            href="/blogs/story/1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
             
            <div
              className="relative"
            >
              <img
                alt="Sample Title 1"
                className="rounded-xl object-cover w-full h-56"
              />
              <div
                className="group-hover:bg-transparent transition rounded-xl duration-300 absolute bottom-0 top-0 right-0 left-0 bg-gray-700 opacity-25"
              />
            </div>
            <div
              className="pb-2 px-2"
            >
              <div
                className="py-2"
              >
                <p
                  className="w-80 text-stone-900 text-xl font-medium capitalize line-clamp-2 group-hover:text-blue-700"
                >
                  Sample Title 1
                </p>
              </div>
              <div
                className="flex justify-between items-center my-1"
              >
                <p
                  className="bg-blue-400 group-hover:bg-blue-500 px-4 py-2 rounded-md text-white font-medium font-Outfit"
                />
                <div
                  className="flex"
                >
                  <span
                    className="text-gray-600 dark:text-white inline-flex items-center lg:ml-auto md:ml-0 ml-auto leading-none text-md"
                  >
                    2023-10-30
                  </span>
                </div>
              </div>
            </div>
          </a>
           
        </div>
      </div>
    </div>
    <div>
      <div
        className="dark:text-white flex flex-col justify-center lg:py-5 md:py-5 overflow-hidden w-80  transition-shadow duration-300 rounded group "
      >
        <div
          className="bg-white rounded-xl"
        >
          <a
            href="/blogs/story/2"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
             
            <div
              className="relative"
            >
              <img
                alt="Sample Title 2"
                className="rounded-xl object-cover w-full h-56"
              />
              <div
                className="group-hover:bg-transparent transition rounded-xl duration-300 absolute bottom-0 top-0 right-0 left-0 bg-gray-700 opacity-25"
              />
            </div>
            <div
              className="pb-2 px-2"
            >
              <div
                className="py-2"
              >
                <p
                  className="w-80 text-stone-900 text-xl font-medium capitalize line-clamp-2 group-hover:text-blue-700"
                >
                  Sample Title 2
                </p>
              </div>
              <div
                className="flex justify-between items-center my-1"
              >
                <p
                  className="bg-blue-400 group-hover:bg-blue-500 px-4 py-2 rounded-md text-white font-medium font-Outfit"
                />
                <div
                  className="flex"
                >
                  <span
                    className="text-gray-600 dark:text-white inline-flex items-center lg:ml-auto md:ml-0 ml-auto leading-none text-md"
                  >
                    2023-10-31
                  </span>
                </div>
              </div>
            </div>
          </a>
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
